### ISSUE TITLE
Nullable warning when tmp variable introduced

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#34800]
**Version Used**: 
3.1.0-beta1-19172-05+edd2de88fb3e84a097fb30b4070e0f219f624e40

**Repro**:
```C#
#nullable enable
using System;

public class Program
{
    public static void Main()
    {
        A? itemA = GetA();

        if (itemA?.Value != null)
        {
            Console.WriteLine(itemA.Name); // no warning
        }

        B? tmp = itemA?.Value;
        if (tmp != null)
        {
            Console.WriteLine(itemA.Name); // warns
        }
    }

    private static A? GetA() => new A();

    class A
    {
        public string Name = "name";
        public B? Value = new B();
    }

    class B
    {
        public string? Value = "value";
    }
}
```

Here the first case guarded by `itemA?.Value != null` doesn't warn, but the second case guarded by the `tmp != null` variable does warn: "Possible dereference of a null reference."

I'm not sure if this is supposed to not warn or if this falls into the same theorem-proving area as https://github.com/dotnet/roslyn/issues/34665.

cc: @cston, @gafter, @dotnet/nullablefc

**Comment by jcouv** [comment]
From discussion with Mads, this requires alias analysis which is out-of-scope for C# 8.

**Comment by jaredpar** [comment]
Closing as we've decided not to do this type of dependent / alias flow analysis in the compiler.
